src/main/frontend/worker/db_worker.cljs

```diff
@@ -442,494 +441,291 @@
+    (p/let [pool (<get-opfs-pool repo)]
+      (<import-db pool data)
+      nil)))
+
+(def-thread-api :search/search-blocks
+  [repo q option]
+  (p/let [search-db (get-search-db repo)
+          conn (worker-state/get-datascript-conn repo)]
+    (search/search-blocks repo conn search-db q option)))
+
+(def-thread-api :search/upsert-blocks
+  [repo blocks]
+  (p/let [db (get-search-db repo)]
+    (search/upsert-blocks! db (bean/->js blocks))
+    nil))
+
+(def-thread-api :search/delete-blocks
+  [repo ids]
+  (p/let [db (get-search-db repo)]
+    (search/delete-blocks! db ids)
+    nil))
+
+(def-thread-api :search/truncate-tables
+  [repo]
+  (p/let [db (get-search-db repo)]
+    (search/truncate-table! db)
+    nil))
+
+(def-thread-api :search/build-blocks-indice
+  [repo]
+  (when-let [conn (worker-state/get-datascript-conn repo)]
+    (search/build-blocks-indice repo @conn)))
+
+(def-thread-api :search/build-pages-indice
+  [_repo]
+  nil)
+
+(def-thread-api :general/apply-outliner-ops
+  [repo ops opts]
+  (when-let [conn (worker-state/get-datascript-conn repo)]
+    (try
+      (worker-util/profile
+       "apply outliner ops"
+       (outliner-op/apply-ops! repo conn ops (worker-state/get-date-formatter repo) opts))
+      (catch :default e
+        (let [data (ex-data e)
+              {:keys [type payload]} (when (map? data) data)]
+          (case type
+            :notification
+            (worker-util/post-message type [(:message payload) (:type payload)])
+            (throw e)))))))
+
+(def-thread-api :general/file-writes-finished?
+  [repo]
+  (let [conn (worker-state/get-datascript-conn repo)
+        writes @file/*writes]
+    ;; Clean pages that have been deleted
+    (when conn
+      (swap! file/*writes (fn [writes]
+                            (->> writes
+                                 (remove (fn [[_ pid]] (d/entity @conn pid)))
+                                 (into {})))))
+    (if (empty? writes)
+      true
+      (do
+        (prn "Unfinished file writes:" @file/*writes)
+        false))))
+
+(def-thread-api :general/page-file-saved
+  [request-id _page-id]
+  (file/dissoc-request! request-id)
+  nil)
+
+(def-thread-api :general/sync-app-state
+  [new-state]
+  (worker-state/set-new-state! new-state)
+  nil)
+
+(def-thread-api :general/sync-ui-state
+  [repo state]
+  (undo-redo/record-ui-state! repo (ldb/write-transit-str state))
+  nil)
+
+(def-thread-api :export/get-debug-datoms
+  [repo]
+  (when-let [db (worker-state/get-sqlite-conn repo)]
+    (let [conn (worker-state/get-datascript-conn repo)]
+      (worker-export/get-debug-datoms conn db))))
+
+(def-thread-api :export/get-all-pages
+  [repo]
+  (when-let [conn (worker-state/get-datascript-conn repo)]
+    (worker-export/get-all-pages repo @conn)))
+
+(def-thread-api :export/get-all-page->content
+  [repo]
+  (when-let [conn (worker-state/get-datascript-conn repo)]
+    (worker-export/get-all-page->content repo @conn)))
+
+(def-thread-api :undo-redo/undo
```

Comment:
Would it be helpful long term to use standard namespace keywords rather than shortened namespace ones? We also have these in shortcut.config and they're harder to work with since they aren't really unique and are context dependent. Semantically they don't feel meaningful since namespaces get created arbitrarily with no specific criteria

Reply:
what about using namespace `thread-api` for all these keywords?

Reply:
update:
has been made to use the namespace "thread-api",
ensuring these keywords won't conflict with others serving different purposes.